TYPO3.Neos/Classes/TYPO3/Neos/TypoScript/DimensionsMenuImplementation.php

```diff
@@ -0,0 +1,216 @@
+                foreach ($targetDimensionsToMatch as $dimensionName => $dimensionValue) {
+                    if ($targetDimensions[$dimensionName] !== $dimensionValue) {
+                        continue 2;
+                    }
+                }
+            }
+
+            $nodeInDimensions = $this->getNodeInDimensions($allowedCombination, $targetDimensions);
+
+            // no match, so we look further...
+            if ($nodeInDimensions === null && $includeAllPresets) {
+                foreach ($allowedCombination[$pinnedDimensionName] as $allowedPresetIdentifier) {
+                    $acceptableCombination = [$pinnedDimensionName => $allDimensionPresets[$pinnedDimensionName]['presets'][$allowedPresetIdentifier]['values']];
+                    $allowedAdditionalPresets = $this->configurationContentDimensionPresetSource->getAllowedDimensionPresetsAccordingToPreselection('country', [$pinnedDimensionName => $allowedPresetIdentifier]);
+                    foreach ($allowedAdditionalPresets as $allowedAdditionalDimensionName => $allowedAdditionalPreset) {
+                        $acceptableCombination[$allowedAdditionalDimensionName] = $allowedAdditionalPreset['presets'][$allowedAdditionalPreset['defaultPreset']]['values'];
+                    }
+                    $nodeInDimensions = $this->getNodeInDimensions($acceptableCombination, []);
+                    if ($nodeInDimensions !== null) {
+                        continue;
+                    }
+                }
+            }
+
+            if ($nodeInDimensions !== null && $this->isNodeHidden($nodeInDimensions)) {
+                $nodeInDimensions = null;
+            }
+
+            // determine metadata for target dimensions of node
+            array_walk($targetDimensions, function (&$dimensionValue, $dimensionName, $allDimensionPresets) use ($pinnedDimensionName) {
+                $dimensionValue = [
+                    'value' => $dimensionValue,
+                    'label' => $allDimensionPresets[$dimensionName]['presets'][$dimensionValue]['label'],
+                    'isPinnedDimension' => ($pinnedDimensionName === null || $dimensionName == $pinnedDimensionName) ? true : false
+                ];
+            }, $allDimensionPresets);
+
+            if ($pinnedDimensionName === null) {
+                $itemLabel = Utility::evaluateEelExpression($this->getLabelExpression(), $this->eelEvaluator, array('node' => $nodeInDimensions), $this->defaultContextConfiguration);
+            } else {
+                $itemLabel = $targetDimensions[$pinnedDimensionName]['label'];
+            }
+
+            $menuItems[] = [
+                'node' => $nodeInDimensions,
+                'state' => $this->calculateItemState($nodeInDimensions),
+                'label' => $itemLabel,
+                'dimensions' => $allowedCombination,
+                'targetDimensions' => $targetDimensions,
+                'pinnedDimensionName' => $pinnedDimensionName
```

Comment:
As the pinned dimension is the same for all menu entries I don't see a use for it in this array, if you need it you can directly access the value in the template...